### PR TITLE
[BEAM-1446] Create with TypeDescriptor

### PR DESCRIPTION
**Pull request by aviemzur**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by aviemzur** [comment]
R: @tgroh

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java [hunk 1]
```diff
@@ -148,7 +149,8 @@
    * Instead, the {@code Coder} is provided via the {@code coder} argument.
    */
   public static <T> Values<T> empty(Coder<T> coder) {
-    return new Values<>(new ArrayList<T>(), Optional.of(coder));
+    return new Values<>(new ArrayList<T>(), Optional.of(coder),
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java]: add `empty(TypeDescriptor<T> type)`?

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CreateTest.java [hunk 2]
```diff
@@ -316,6 +318,35 @@ public void testCreateGetName() {
...
+    Coder<Integer> defaultCoder = values.getDefaultOutputCoder(pBegin);
+    assertThat(defaultCoder, equalTo(coder));
+  }
+
+  @Test
+  public void testCreateDefaultOutputCoderUsingTypeDescriptor() throws Exception {
+    Coder<Integer> coder = VarIntCoder.of();
+    PBegin pBegin = PBegin.in(p);
+    Create.Values<Integer> values =
+        Create.of(1, 2, 3, 4, 5, 6, 7, 8).withType(new TypeDescriptor<Integer>() {});
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CreateTest.java]: These would be best as something that can't use the default coder inference (e.g. `Record` and `Record2` which appear later in the test. You may have to write a `RecordCoder` that doesn't do anything.

**Comment by aviemzur** [comment]
@tgroh addressed your comments in my last commit.

**Comment by tgroh** [comment]
retest this please

**Comment by aviemzur** [comment]
Retest this please.

**Pull request closed without merging**